ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -151,8 +151,8 @@ ORK_CLASS_AVAILABLE
 
 + (ORKTextAnswerFormat *)textAnswerFormatWithMaximumLength:(NSInteger)maximumLength;
 
-+ (ORKTextAnswerFormat *)textAnswerFormatWithValidationRegex:(NSString *)validationRegex
-                                              invalidMessage:(NSString *)invalidMessage;
++ (ORKTextAnswerFormat *)textAnswerFormatWithValidationRegularExpression:(NSRegularExpression *)validationRegularExpression
```

Comment:
I don't feel bad breaking backwards compatibility on `ResearchKit` in pursuit of a cleaner API.

But if the consensus is that adding back the deprecated initializer is a good idea, I'll happily add it. :)